Aspire Budgeting/Views/SearchBar.swift

```diff
@@ -0,0 +1,66 @@
+//
+// SearchBar.swift
+// Aspire Budgeting
+//
+
+import SwiftUI
+
+struct SearchBar: View {
+  @Binding var text: String
+
+  @State private var isEditing = false
+
+    var body: some View {
+      HStack {
+        TextField("Search ...", text: $text)
+                        .padding(7)
+                        .padding(.horizontal, 25)
+                        .background(Color(.systemGray6))
+                        .cornerRadius(8)
+          .overlay(
+              HStack {
+                  Image(systemName: "magnifyingglass")
+                      .foregroundColor(.gray)
+                      .frame(minWidth: 0, maxWidth: .infinity, alignment: .leading)
+                      .padding(.leading, 8)
+
+                  if isEditing {
+                      Button(action: {
+                          self.text = ""
+                      }) {
+                          Image(systemName: "multiply.circle.fill")
+                              .foregroundColor(.gray)
+                              .padding(.trailing, 8)
+                      }
+                  }
+              }
+          )
+                        .padding(.horizontal, 10)
+                        .onTapGesture {
+                            self.isEditing = true
+                        }
+
+        if isEditing {
+                        Button(action: {
+                            self.isEditing = false
+                            self.text = ""
+                          UIApplication
+                            .shared
+                            .sendAction(#selector(UIResponder.resignFirstResponder), to: nil, from: nil, for: nil)
+
+                        }) {
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/aspirebudgetingmobile/aspirebudgeting_ios/blob/ff44934e95fcfbf9892d0e49040b4320aa9b4509/Aspire Budgeting/Views/SearchBar.swift#L51"></span>Trailing closure syntax should not be used when passing more than one closure argument.<br />
<code>multiple_closures_with_trailing_closure</code> <code>SearchBar.swift:51</code></td>
    </tr>
  </tbody>
</table>